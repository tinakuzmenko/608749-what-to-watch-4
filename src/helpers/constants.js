export const MAX_SHOWN_GENRES = 10;

export const Pages = {
  MAIN: `main`,
  MOVIE: `movie`,
};

export const NavTabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const ALL_GENRES = `All genres`;

export const SHOWN_MOVIES = 8;
export const MAX_SHOWN_MOVIES_LIKE_THIS = 4;
