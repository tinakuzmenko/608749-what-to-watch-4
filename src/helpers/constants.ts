export const ALL_GENRES = `All genres`;
export const TEXTAREA_COLOR = `rgba(255, 255, 255, 0.3)`;
export const MAX_SHOWN_GENRES = 10;
export const RATINGS_QUANTITY = 5;
export const SHOWN_MOVIES = 8;
export const MAX_SHOWN_MOVIES_LIKE_THIS = 4;

export enum Pages {
  MAIN = `main`,
  MOVIE = `movie`,
  VIDEO = `video`,
  SIGN_IN = `sign in`,
  ADD_REVIEW = `add review`,
  MY_LIST = `my list`,
}

export enum NavTabs {
  OVERVIEW = `Overview`,
  DETAILS = `Details`,
  REVIEWS = `Reviews`,
}

export enum smallVideoPlayer {
  WIDTH = `280`,
  HEIGHT = `175`,
}

export enum TimeInSeconds {
  MINUTE = 60,
}

export enum AuthorizationStatus {
  AUTH = `AUTH`,
  NO_AUTH = `NO_AUTH`,
}

export enum Error {
  UNAUTHORIZED = 401,
}

export enum Review {
  MIN_LENGTH = 50,
  MAX_LENGTH = 400,
}

export enum AppRoute {
  MAIN = `/`,
  SIGN_IN = `/login`,
  MY_LIST = `/mylist`,
  MOVIE = `/films`,
  PLAYER = `/player`,
}

export enum Favorites {
  ADD = 1,
  REMOVE = 0,
}

interface ApiConfigInterface {
  URL: string;
  TIMEOUT: number;
  COOKIES: true;
}

export const ApiConfig: ApiConfigInterface = {
  URL: `https://4.react.pages.academy/wtw`,
  TIMEOUT: 5000,
  COOKIES: true
};

export const reviewSubmitButton: {
  [key: string]: string;
} = {
  post: `Post`,
  sending: `Sending...`,
};
