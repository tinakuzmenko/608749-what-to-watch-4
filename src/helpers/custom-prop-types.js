import PropTypes from 'prop-types';

export const CustomPropTypes = {
  MOVIE: PropTypes.shape({
    title: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    date: PropTypes.number.isRequired,
    background: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    votes: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    runTime: PropTypes.number.isRequired,
    preview: PropTypes.string.isRequired,
    videoLink: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    backgroundColor: PropTypes.string.isRequired,
  }).isRequired,

  REVIEW: PropTypes.shape({
    id: PropTypes.number.isRequired,
    user: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
  }).isRequired,
};
