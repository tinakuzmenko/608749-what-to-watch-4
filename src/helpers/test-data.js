export const movie = {
  title: `Snatch`,
  genre: `Crime`,
  date: `2000`,
  background: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  id: 123890,
  description: [`Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`],
  rating: `8,3`,
  ratingDescription: `Good`,
  votes: 1500,
  director: `Guy Ritchie`,
  starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  runTime: `1h 44m`,
};

export const movies = [
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    date: `2018`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/bohemian-rhapsody.jpg`,
    id: 178345,
    description: [`The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`],
    rating: `8,0`,
    ratingDescription: `Good`,
    votes: 800,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    runTime: `2h 14m`,
  },
  {
    title: `Aviator`,
    genre: `Drama`,
    date: `2004`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/aviator.jpg`,
    id: 167456,
    description: [`A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`],
    rating: `7,5`,
    ratingDescription: `Normal`,
    votes: 1650,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    runTime: `2h 50m`,
  }];

export const currentMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  id: 190123,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: `8,9`,
  ratingDescription: `Very good`,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  runTime: `1h 39m`,
};

export const reviews =
[{
  movie: `Snatch`,
  reviews: [{
    author: `Kate Muir`,
    rating: `8,9`,
    date: `December 24, 2019`,
    content: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    id: 345234523,
  },
  {
    author: `Bill Goodykoontz`,
    rating: `8,0`,
    date: `November 18, 2015`,
    content: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    id: 22345234,
  },
  {
    author: `Amanda Greever`,
    rating: `8,0`,
    date: `November 18, 2015`,
    content: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    id: 4563456345,
  },
  {
    author: `Matthew Lickona`,
    rating: `7,2`,
    date: `December 20, 2016`,
    content: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    id: 3456345437,
  }]
},
{
  movie: `The Grand Budapest Hotel`,
  reviews: [{
    author: `Kate Muir`,
    rating: `8,9`,
    date: `December 24, 2019`,
    content: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    id: 345234523,
  },
  {
    author: `Bill Goodykoontz`,
    rating: `8,0`,
    date: `November 18, 2015`,
    content: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    id: 22345234,
  },
  {
    author: `Amanda Greever`,
    rating: `8,0`,
    date: `November 18, 2015`,
    content: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    id: 4563456345,
  },
  {
    author: `Matthew Lickona`,
    rating: `7,2`,
    date: `December 20, 2016`,
    content: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    id: 3456345437,
  }]
}];
