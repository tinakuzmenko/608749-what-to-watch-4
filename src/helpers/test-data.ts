import {MovieInterface, ReviewInterface} from '../types';

export const noop = () => {
  // Mock function for test props
};

export const movie: MovieInterface = {
  title: `Snatch`,
  genre: `Crime`,
  date: 2000,
  background: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  posterImage: `img/snatch.jpg`,
  id: 123890,
  description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`,
  rating: 8.3,
  votes: 1500,
  director: `Guy Ritchie`,
  starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  runTime: 104,
  videoLink: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const movies: Array<MovieInterface> = [
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    date: 2018,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/bohemian-rhapsody.jpg`,
    posterImage: `img/bohemian-rhapsody.jpg`,
    id: 178345,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`,
    rating: 8.0,
    votes: 800,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    runTime: 134,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  },
  {
    title: `Aviator`,
    genre: `Drama`,
    date: 2004,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/aviator.jpg`,
    posterImage: `img/aviator.jpg`,
    id: 167456,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    rating: 7.5,
    votes: 1650,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    runTime: 170,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  },
];

export const currentMovie: MovieInterface = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: 2014,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  id: 190123,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: 8.9,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  runTime: 99,
  videoLink: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const reviews: Array<ReviewInterface> =
[{
  id: 4563456345,
  user: {
    id: 1223,
    name: `Amanda Greever`,
  },
  rating: 8.0,
  comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  date: `2019-05-08T14:13:56.569Z`,
},
{
  id: 123124,
  user: {
    id: 12415252,
    name: `Amanda Greever`,
  },
  rating: 7.0,
  comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  date: `2019-05-08T14:13:56.569Z`,
},
{
  id: 23462324631461,
  user: {
    id: 1234,
    name: `Amanda Greever`,
  },
  rating: 8.0,
  comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  date: `2019-05-08T14:13:56.569Z`,
}];

export const genres: Array<string> = [`All genres`, `Drama`, `Thriller`, `Comedy`];

export const allGenresMovies: Array<MovieInterface> = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Kids & Family`,
    date: 2018,
    backgroundColor: `#FFFFFF`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    id: 189234,
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    rating: 6.6,
    votes: 500,
    director: `David Yates`,
    starring: [`Johnny Depp`, `Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 133,
    isFavorite: true,
  },
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    date: 2018,
    backgroundColor: `#FFFFFF`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/bohemian-rhapsody.jpg`,
    posterImage: `img/bohemian-rhapsody.jpg`,
    id: 178345,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`,
    rating: 8.0,
    votes: 800,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/f/fa/Tornado_time_lapse.webm`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 134,
    isFavorite: false,
  },
  {
    title: `Aviator`,
    genre: `Drama`,
    date: 2004,
    backgroundColor: `#FFFFFF`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/aviator.jpg`,
    posterImage: `img/aviator.jpg`,
    id: 167456,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    rating: 7.5,
    votes: 1650,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/9/9b/St._Lambertus_%28Immerather_Dom%29_2016.webm`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 170,
    isFavorite: false,
  },
  {
    title: `Shutter Island`,
    genre: `Thriller`,
    date: 2010,
    backgroundColor: `#FFFFFF`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/shutter-island.jpg`,
    posterImage: `img/shutter-island.jpg`,
    id: 156567,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`,
    rating: 8.1,
    votes: 900,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/4/4d/Baha%27i_Temple_--_Wilmette_%2C_IL_--_Drone_Video_%28DJI_Spark%29.webm`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 139,
    isFavorite: false,
  },
  {
    title: `Pulp Fiction`,
    genre: `Crime`,
    date: 1994,
    backgroundColor: `#FFFFFF`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/pulp-fiction.jpg`,
    posterImage: `img/pulp-fiction.jpg`,
    id: 145678,
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    rating: 8.9,
    votes: 2500,
    director: `Quentin Tarantino`,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`, `Bruce Willis`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/9/95/Aerial_View_of_The_Vessel_at_Hudson_Yards%2C_New_York_City.webm`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 178,
    isFavorite: false,
  },
  {
    title: `No Country for Old Men`,
    genre: `Thriller`,
    date: 2007,
    backgroundColor: `#FFFFFF`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/no-country-for-old-men.jpg`,
    posterImage: `img/no-country-for-old-men.jpg`,
    id: 134789,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    rating: 8.1,
    votes: 870,
    director: `Ethan Coen, Joel Coen`,
    starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 123,
    isFavorite: false,
  },
  {
    title: `Snatch`,
    genre: `Crime`,
    date: 2000,
    backgroundColor: `#FFFFFF`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    posterImage: `img/snatch.jpg`,
    id: 123890,
    description: `Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`,
    rating: 8.3,
    votes: 1500,
    director: `Guy Ritchie`,
    starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/d/d0/Un_Hiver_%C3%A0_Paris_-_Vimeo.webm`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 164,
    isFavorite: false,
  }
];
