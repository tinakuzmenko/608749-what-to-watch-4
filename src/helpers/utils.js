export const extend = (state, newStateValue) => {
  return Object.assign({}, state, newStateValue);
};

export const getMoviesReviews = (allReviews, movie) => {
  return allReviews.filter((movieReviews) => movieReviews.movie === movie.title);
};

export const sliceReviews = (reviews) => {
  const sliceIndex = Math.ceil(reviews.length / 2);
  const firstColReviews = reviews.slice(0, sliceIndex);
  const secondColReviews = reviews.slice(sliceIndex, reviews.length);

  return [firstColReviews, secondColReviews];
};

export const getRatingLevel = (rating) => {
  if (rating >= 0 && rating < 3) {
    return `Bad`;
  }

  if (rating >= 3 && rating < 5) {
    return `Normal`;
  }

  if (rating >= 5 && rating < 8) {
    return `Good`;
  }

  if (rating >= 8 && rating < 10) {
    return `Very good`;
  }

  if (rating === 10) {
    return `Awesome`;
  }

  return null;
};

export const getRunTimeFormat = (runTime) => {
  const hours = Math.trunc(runTime / 60);
  const minutes = runTime - (hours * 60);

  return `${hours}h ${minutes}m`;
};

export const getRatingFormat = (rating) => {
  let formattedRating;

  if (rating % 1 !== 0) {
    formattedRating = rating.toString().split(`.`).join(`,`);
  } else {
    formattedRating = `${rating.toString()},0`;
  }

  return formattedRating;
};
