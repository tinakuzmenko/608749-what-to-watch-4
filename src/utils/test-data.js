export const movie = {
  title: `Snatch`,
  genre: `Crime`,
  date: `2000`,
  background: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  id: 123890,
  description: [`Unscrupulous boxing promoters, violent bookmakers, a Russian gangster, incompetent amateur robbers and supposedly Jewish jewelers fight to track down a priceless stolen diamond.`],
  rating: `8,3`,
  ratingDescription: `Good`,
  votes: 1500,
  director: `Guy Ritchie`,
  starring: [`Jason Statham`, `Brad Pitt`, `Benicio Del Toro`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
};

export const movies = [
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    date: `2018`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/bohemian-rhapsody.jpg`,
    id: 178345,
    description: [`The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`],
    rating: `8,0`,
    ratingDescription: `Good`,
    votes: 800,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  },
  {
    title: `Aviator`,
    genre: `Drama`,
    date: `2004`,
    background: `https://placeimg.com/1300/512/nature`,
    poster: `img/aviator.jpg`,
    id: 167456,
    description: [`A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`],
    rating: `7,5`,
    ratingDescription: `Normal`,
    votes: 1650,
    director: `Martin Scorsese`,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  }];

export const currentMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  date: `2014`,
  background: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  id: 190123,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: `8,9`,
  ratingDescription: `Very good`,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`
};
