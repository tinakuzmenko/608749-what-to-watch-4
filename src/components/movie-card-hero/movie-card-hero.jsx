import React from 'react';
import PropTypes from 'prop-types';
import {CustomPropTypes} from '../../helpers/custom-prop-types';
import PageHeader from '../page-header/page-header';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../store/user/selectors';
import {AuthorizationStatus, AppRoute, Pages} from '../../helpers/constants';
import {Link} from 'react-router-dom';
import MyListButton from '../my-list-button/my-list-button';

const MovieCardHero = ({currentMovie, isSignedIn}) => {
  const addReviewButton = (
    <Link
      to={`${AppRoute.MOVIE}/${currentMovie.id}/review`}
      className="btn movie-card__button"
    >Add review
    </Link>
  );

  return (
    <div className="movie-card__hero">
      <div className="movie-card__bg">
        <img src={currentMovie.background} alt={currentMovie.title} />
      </div>
      <h1 className="visually-hidden">WTW</h1>
      <PageHeader
        currentPage={Pages.MOVIE}
      />
      <div className="movie-card__wrap">
        <div className="movie-card__desc">
          <h2 className="movie-card__title">{currentMovie.title}</h2>
          <p className="movie-card__meta">
            <span className="movie-card__genre">{currentMovie.genre}</span>
            <span className="movie-card__year">{currentMovie.date}</span>
          </p>
          <div className="movie-card__buttons">
            <Link
              className="btn btn--play movie-card__button"
              to={`${AppRoute.PLAYER}/${currentMovie.id}`}
            >
              <svg viewBox="0 0 19 19" width="19" height="19">
                <use xlinkHref="#play-s"></use>
              </svg>
              <span>Play</span>
            </Link>
            <MyListButton
              movie={currentMovie}
            />
            {isSignedIn && addReviewButton}
          </div>
        </div>
      </div>
    </div>
  );
};

MovieCardHero.propTypes = {
  currentMovie: CustomPropTypes.MOVIE,
  isSignedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  isSignedIn: getAuthorizationStatus(state) === AuthorizationStatus.AUTH,
});

export default connect(mapStateToProps)(MovieCardHero);
