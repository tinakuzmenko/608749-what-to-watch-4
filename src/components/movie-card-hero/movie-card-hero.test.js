import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {MovieCardHero} from './movie-card-hero';
import {movie} from '../../helpers/test-data';

describe(`MovieCardHero`, () => {
  const mockStore = configureStore([]);

  it(`Should render correctly`, () => {
    const store = mockStore({
      currentMovie: movie,
      currentPage: `movie`,
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <MovieCardHero
              currentMovie={movie}
              onPlayButtonClick={() => {}}
            />
          </Provider>, {
            createNodeMock: () => {
              return {};
            }
          })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
