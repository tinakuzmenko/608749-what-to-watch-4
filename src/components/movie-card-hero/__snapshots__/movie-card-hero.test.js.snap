// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardHero Should render correctly 1`] = `
<div
  className="movie-card__hero"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="Snatch"
      src="https://placeimg.com/1300/512/nature"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        href="/mylist"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="asdasd"
            height="63"
            src="https://4.react.pages.academy/wtw/asda.jpg"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__desc"
    >
      <h2
        className="movie-card__title"
      >
        Snatch
      </h2>
      <p
        className="movie-card__meta"
      >
        <span
          className="movie-card__genre"
        >
          Crime
        </span>
        <span
          className="movie-card__year"
        >
          2000
        </span>
      </p>
      <div
        className="movie-card__buttons"
      >
        <a
          className="btn btn--play movie-card__button"
          href="/player/123890"
          onClick={[Function]}
        >
          <svg
            height="19"
            viewBox="0 0 19 19"
            width="19"
          >
            <use
              xlinkHref="#play-s"
            />
          </svg>
          <span>
            Play
          </span>
        </a>
        <button
          className="btn btn--list movie-card__button"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="20"
            viewBox="0 0 19 20"
            width="19"
          >
            <use
              xlinkHref="#add"
            />
          </svg>
          <span>
            My list
          </span>
        </button>
        <a
          className="btn movie-card__button"
          href="/films/123890/review"
          onClick={[Function]}
        >
          Add review
        </a>
      </div>
    </div>
  </div>
</div>
`;
